docs/Personalities.md

```diff
@@ -0,0 +1,275 @@
+   the API, for example:
+   - What submission formats are accepted?
+   - What cryptographic promises/guarantees does the personality generate?
+   - Which Merkle tree primitives are exposed to external users?
+
+
+### Storage
+
+A Trillian personality may need an additional storage mechanism over and above
+the Merkle tree storage provided by the Trillian core services.  For example,
+if the Trillian core is run by a separate operator, then
+[audit verification data may need to be stored](#auditability).
+
+However, note that the Trillian core APIs allow a personality to store
+additional data that is associated with a leaf in the underlying Merkle tree,
+in the `ExtraData` field.  This extra data is immutable but is not covered by
+the Merkle tree hashing algorithms, and so cannot be used for cryptographic
+purposes.
+
+
+### Traffic Control
+
+The core Trillian service can be configured to provide traffic control limits
+using its [quota system](../quota/doc.go); out of the box, this can be governed
+by:
+ - global rate limits (for both read and write operations)
+ - per-Merkle tree rate limits (for read and write).
+
+A particular application may also want to provide other kinds of limits.  This
+is supported by the per-*user* quota limits, where the *user* is entirely
+specified by the Trillian personality, using the `ChargeTo` field in each core
+API request.
+
+A typical use case for this is to add a `ChargeTo.User` string that identifies
+an external requestor, so traffic from a single source can be limited without
+affecting other users.
+
+However, the user quota system can also be used for more flexible limits
+&ndash; for example, by applying limits to particular authentication keys.
+
+
+### Monitoring
+
+A personality that is in real-world use should probably be monitored, for
+performance, alerting and debugging.  The core Trillian services use a
+[metrics interface](../monitoring/metrics.go) to support this, which can also
+be re-used in an associated personality.
+
+## Examples
+
+This section describes two Trillian personalities, together with their choices
+for the various design decisions described above.
+
+### Certificate Transparency
+
+Certificate Transparency (CT) is the original transparency application, and has
+been successful in improving the security of the web PKI ecosystem.  It is also
+the most mature and widely deployed Trillian use case, built on the
+[CTFE personality](https://github.com/google/certificate-transparency-go/blob/master/trillian/ctfe).
+
+The [conceptual responsibilities](#conceptual-responsibilities) of the CTFE
+personality are given by:
+
+ - Admission Control: Submissions have to take the form of a chain of valid
+   X.509 certificates, each certificate signed by the next certificate in the
+   chain, until a acceptable root certificate is reached.  The set of accepted
+   roots is configured in the personality at start-of-day.
```

Comment:
/start-of-day/start up/?
"start-of-day" might be mis-interpreted to mean "automatically each morning"